src/analysis/has_public_example.d

```diff
@@ -0,0 +1,234 @@
+// Distributed under the Boost Software License, Version 1.0.
+//    (See accompanying file LICENSE_1_0.txt or copy at
+//          http://www.boost.org/LICENSE_1_0.txt)
+
+module analysis.has_public_example;
+
+import analysis.base;
+import dsymbol.scope_ : Scope;
+import dparse.ast;
+import dparse.lexer;
+
+import std.stdio;
+
+/**
+ * Checks for public declarations without a documented unittests.
+ * For now, variable and enum declarations aren't checked.
+ */
+class HasPublicExampleCheck : BaseAnalyzer
+{
+	alias visit = BaseAnalyzer.visit;
+
+	this(string fileName, const(Scope)* sc, bool skipTests = false)
+	{
+		super(fileName, sc, skipTests);
+	}
+
+    override void visit(const Module mod)
+    {
+        Declaration lastDecl;
+        bool hasPublicUnittest = true; // reset on lastDecl assignment
```

Comment:
I'd rather name this variable `hasNoPublicTest` and change the related relational expressions, because

- then you'll can remove the initializer.
- if you look at the use spots, you always negate the variable.

hence you'll have the same meaning with less complexity.